common/src/com/thoughtworks/go/serverhealth/ServerHealthService.java

```diff
@@ -72,6 +73,25 @@ public HealthStateType update(ServerHealthState serverHealthState) {
         }
     }
 
+    // called from spring timer
+    public synchronized void onTimer(CruiseConfigProvider provider) {
+        CruiseConfig currentConfig = provider.getCurrentConfig();
+        getAllValidLogs(currentConfig);
+        LOG.info("Recomputing material to pipeline mappings.");
+
+        HashMap<ServerHealthState, Set<String>> erroredPipelines = new HashMap<>();
+
+        for (Map.Entry<HealthStateType, ServerHealthState> entry : serverHealth.entrySet()) {
+            erroredPipelines.put(entry.getValue(), entry.getValue().getPipelineNames(currentConfig));
+        }
+        pipelinesWithErrors = erroredPipelines;
+        LOG.info("Done recomputing material to pipeline mappings.");
+    }
+
+    public Set<String> getPipelinesWithErrors(ServerHealthState serverHealthState) {
+        return pipelinesWithErrors.get(serverHealthState);
```

Comment:
Since this is not synced and not got in one shot, it's possible that stale data will be mixed with new data (while it is being called multiple times from the ERB file). I suppose that's ok?